pkg/services/oracle/network.go

```diff
@@ -89,6 +89,10 @@ func getDefaultClient(cfg config.OracleConfiguration) *http.Client {
 	// Override default CheckRedirect behaviour so that there's no restriction on the
 	// number of redirects (10 at max is allowed by default).
 	client.CheckRedirect = func(req *http.Request, via []*http.Request) error {
+		if len(via) > 0 && via[0].URL.Scheme == "https" && req.URL.Scheme != "https" {
```

Comment:
@roman-khimov, it worked like this in C# before https://github.com/neo-project/neo-modules/pull/692 merge (see https://docs.microsoft.com/en-us/dotnet/api/system.net.http.httpclienthandler.allowautoredirect?view=net-6.0#remarks). But I'm not sure it behaves like this now.